### ISSUE TITLE
Display of PrincipalPubl Info

### ISSUE THREAD

**Issue by paregorios** [DCLP/dclpxsltbox#123]
Originally opened by @rla2118 [on another tracker](https://github.com/paregorios/dclpxsltbox/issues/14) as follows:

Publication info is absent from HTML (see, e.g., https://dclp.github.io/dclpxsltbox/output/dclp/129/128967.html). Please add it, drawing on (https://github.com/DCLP/idp.data/blob/dclp/DCLP/129/128967.xml#l99)

I would suggest an HTML display similar to what we get in HGV. In the case of the above papyrus, it would look like: P.Oxy. 75 5040

Also, I think it should appear at or near the top of the metadata, perhaps just below Work?

**Comment by paregorios** [comment]
This is probably just an example of what we aimed to resolve with #109. @cgajjar05 please address that issue (which is now unblocked) and then circle back to this ticket to make sure it is resolved. Meantime, I'm marking this ticket "blocked" pending resolution of #109.

**Comment by jcowey** [comment]
There are two types of data to be dealt with here.

The xpath is always

../<div type="bibliography" subtype="principalEdition">

When the path is

<div type="bibliography" subtype="principalEdition">/<listBibl>/<bibl type="publication" subtype="principal">

then concatenate all text nodes with \s{1} between each one until </bibl>

e.g.
https://github.com/DCLP/idp.data/blob/dclp/DCLP/113/112358.xml#l128-#l136

I would expect to see:

Principal Edition 	P. Oxy. 71 4803

When the path is

<div type="bibliography" subtype="principalEdition">/<listBibl>/<bibl type="reference" subtype="principal">

then using the XSLT that is used to create the Provisional papyri.info output in

http://papyri.info/biblio/75990

create from 

https://github.com/DCLP/idp.data/blob/dclp/Biblio/76/75990.xml

the output remembering to cater for

https://github.com/DCLP/idp.data/blob/dclp/Biblio/76/75990.xml#l9

(this is not currently done by the XSLT for biblio in papyri.info). It would be nice if DCLP XSLT could then be made available to improve the papyri.info XSLT

e.g.
https://github.com/DCLP/idp.data/blob/dclp/DCLP/109/108908.xml#l78-#l92

I would expect to see:

Reference Edition 	John LUNDON, "PSI inv. 320 verso: Fragment einer Rede?," Zeitschrift fur Papyrologie und Epigraphik, 159 (2007) 56-60.

**Comment by cgajjar05** [comment]
@jcowey : We already have one title "Reference Edition" for subtype = "referenceEdition". So, what do you suggest for that?

For your reference check here code of metadata.xsl : 
https://github.com/DCLP/navigator/blob/xslt-master/pn-xslt/metadata.xsl#L778

**Comment by jcowey** [comment]
Here is a list of the different types of bibliography:

```XML
<div type="bibliography" subtype="principalEdition">
    <listBibl>
        <bibl type="publication" subtype="principal">...</bibl>
        <bibl type="reference" subtype="principal">...</bibl>
        <bibl type="reference" subtype="partial">...</bibl>
        <bibl type="reference" subtype="previous">...</bibl>
        <bibl type="reference" subtype="readings">...</bibl>
    </bibl>
</div>
```

**Comment by jcowey** [comment]
The last comments in #107 
gave the commits which brought about the change to all bibliography.
https://github.com/DCLP/idp.data/commit/2a66eb239ffaf5fbef7e760eb46ea84d0aba5ed4

Perhaps this entails slight reworking of the XSLT. I hope this is sufficient info.

**Comment by cgajjar05** [comment]
Ok, but we are planning to give title "Reference Edition" for <bibl type="reference" subtype="principal">...</bibl>
and we already have one "Reference Edition" title for type="bibliography" subtype="referenceEdition"

So, I guess we need one different title for <bibl type="reference" subtype="principal">...</bibl>

**Comment by paregorios** [comment]
@jcowey I think the question is: did we entirely get rid of the encoding @cgajjar05 is describing?

**Comment by jcowey** [comment]
@paregorios and @cgajjar05 - I am not entirely sure, but I think the answer is yes. All bibliography has been change to the new style as given in the fictional entry in #107

**Comment by paregorios** [comment]
@cgajjar05 can you do a pull on the repository and then try using grep to see if any of the old-style encoding remains, or if it has all been replaced?

**Comment by jcowey** [comment]
As you may well have noticed:
type="bibliography" subype="referenceEdition" 
does not exist any more in 
https://github.com/DCLP/idp.data/tree/dclp/DCLP

**Comment by paregorios** [comment]
Great. @cgajjar05 is it clear now how to proceed?

**Comment by cgajjar05** [comment]
yes, perfect. Thanks both of you.

**Comment by paregorios** [comment]
This ticket now supersedes #109, which I am going to mark "won't fix" and close. Removing block label on this ticket.

**Comment by cgajjar05** [comment]
@jcowey : Can you please explain from where will I get information about second author in the examples that you pointed out?. Referencing your previous comment on this thread : https://github.com/DCLP/dclpxsltbox/issues/123#issuecomment-86093997

**Comment by jcowey** [comment]
When I wrote:

"then using the XSLT that is used to create the Provisional papyri.info output in

http://papyri.info/biblio/75990

create from

https://github.com/DCLP/idp.data/blob/dclp/Biblio/76/75990.xml"

I meant use the code (I am pretty sure it is this one)

https://github.com/papyri/navigator/blob/master/pn-xslt/Biblio2HTML.xsl

the output

"remembering to cater for

https://github.com/DCLP/idp.data/blob/dclp/Biblio/76/75990.xml#l9

(this is not currently done by the XSLT for biblio in papyri.info). It would be nice if DCLP XSLT could then be made available to improve the papyri.info XSLT

please also include a link to the biblio record - it would be best if this were a clickable link at the end of the string"

Does this help?

**Comment by jcowey** [comment]
https://github.com/DCLP/idp.data/blob/dclp/DCLP/109/108908.xml#l81
is where you find the biblio record that you have to go to

**Comment by cgajjar05** [comment]
@jcowey  : Thank you for the comment, but I am still in confusion. I still don't have any file to get content "Zeitschrift fur Papyrologie und Epigraphik, ". I can see that this is available in output file :  http://papyri.info/biblio/75990. But what would be the xml? I tried to click on the "xml" hyperlink on output page (http://papyri.info/biblio/75990).

**Comment by paregorios** [comment]
@cgajjar05 the XML is housed in a parallel directory inside idp.data, and uses a similar numbering convention. So if the biblio ID is 75990, you'd want to pull the XML from https://github.com/DCLP/idp.data/blob/dclp/Biblio/76/75990.xml . This will need to happen locally on the filesystem, I assume. Look through the XSL code for other examples of the use of the xpath "document()" function to see if you can find an example.

**Comment by cgajjar05** [comment]
Thank you @paregorios  for that. But I am still in search for the required text, i.e. information about second author ("Zeitschrift fur Papyrologie und Epigraphik, "). I couldn't find these in the specified xml too.

**Comment by paregorios** [comment]
ah, I think the question is over to @jcowey to help us sort out whether this is a deficiency in the XML or in the example or something else.

**Comment by cgajjar05** [comment]
Ok, thank you for your time. I can see the required texts in http://papyri.info/biblio/75990. So, it must be somewhere else.

**Comment by jcowey** [comment]
OK, let me try to explain. A journal article and an article in a collection of articles "appearsIn" a journal / book made up of articles written by different people and edited by an editor (or number of editors). In the XML of each relevant entry of "Biblio"
https://github.com/DCLP/idp.data/tree/dclp/Biblio
we make a statement about the journal / book in which the relevant article appears in. In the case of https://github.com/DCLP/idp.data/blob/dclp/Biblio/76/75990
the information is to be found in 
https://github.com/DCLP/idp.data/blob/dclp/Biblio/76/75990.xml#l12-19
this tells you that the article in question appears in ZPE
https://github.com/DCLP/idp.data/blob/dclp/Biblio/76/75990.xml#l16
in code line 14 the biblio number of the biblio entry for the journal "ZPE" is stored in
https://github.com/DCLP/idp.data/blob/dclp/Biblio/76/75990.xml#l14
this will direct you to 
https://github.com/DCLP/idp.data/blob/dclp/Biblio/1/511.xml
here in code line 3
https://github.com/DCLP/idp.data/blob/dclp/Biblio/1/511.xml#l3
you will find 
<title level="j" type="main">...
as the place that you will find what you should include to produce the HTML presentation

**Comment by cgajjar05** [comment]
@jcowey  :  I have one doubt in Reference Edition.
Why two authors are separated here : 

https://github.com/DCLP/idp.data/blob/dclp/DCLP/109/108908.xml#l78-#l92

Reference Edition John LUNDON, "PSI inv. 320 verso: Fragment einer Rede?," Zeitschrift fur Papyrologie und Epigraphik, 159 (2007) 56-60. biblio link

But here, two authors are represented different way 

https://github.com/DCLP/idp.data/blob/dclp/DCLP/33/32950.xml#l73-#l92

Reference Edition Robert W. DANIEL and Franco MALTOMINI, Supplementum Magicum. Vol. I (Suppl. Mag. I)., (Opladen 1990). biblio link

Request you to please help.

**Comment by jcowey** [comment]
@jcowey  :  I have one doubt in Reference Edition.
Why two authors are separated here : 

https://github.com/DCLP/idp.data/blob/dclp/DCLP/109/108908.xml#l78-#l92

Reference Edition John LUNDON, "PSI inv. 320 verso: Fragment einer Rede?," Zeitschrift fur Papyrologie und Epigraphik, 159 (2007) 56-60. biblio link

There is only one author here. Do not fully understand your question.
I am sure that you know this, but the content of 
https://github.com/DCLP/idp.data/blob/dclp/DCLP/109/108908.xml#l82-#l89
is only there to make the xml slightly more understandable. For actual display all the information has to come from the relevant biblio files

**Comment by jcowey** [comment]
But here, two authors are represented different way

https://github.com/DCLP/idp.data/blob/dclp/DCLP/33/32950.xml#l73-#l92

Reference Edition Robert W. DANIEL and Franco MALTOMINI, Supplementum Magicum. Vol. I (Suppl. Mag. I)., (Opladen 1990). biblio link

In this case there are two authors. 
https://github.com/papyri/idp.data/blob/master/Biblio/13/12016.xml#l8-15
and it is from here that you should be getting the information.
Again just to make sure, the content of 
https://github.com/DCLP/idp.data/blob/dclp/DCLP/33/32950.xml#l82-#l89
is only there for guidance. You should not be trying ot extract information from here

**Comment by jcowey** [comment]
Please feel free to contact Carmen on this, if you continue to have difficulties.

**Comment by paregorios** [comment]
blocked for #129

**Comment by paregorios** [comment]
Not blocked for #129 after all (see comment there). I'm taking ownership of this ticket for the moment.

**Comment by paregorios** [comment]
addressed in a series of commits, culminating in https://github.com/DCLP/navigator/commit/aefdb51b0b40c84fef9a5de76daa65124947d395 ... moving status to testing

**Comment by paregorios** [comment]
@jcowey please inspect at your leisure latest transforms at http://dclp.github.io/dclpxsltbox/ to verify completeness of implementation etc.

**Comment by jcowey** [comment]
Finished.
